### ISSUE TITLE
[EN] Error loading page after clicking "Mon entreprise"

### ISSUE THREAD

**Issue by Valdnet** [baimard/gestion#11]
After clicking "Mon entreprise" an internal server error occurred.

![obraz](https://user-images.githubusercontent.com/47037905/128350703-9c8a3fc5-f2f9-418a-8c26-b8447e09702a.png)

Logi:

```
Error	index	Exception: Wrong parameters for Doctrine\DBAL\Driver\PDO\Exception([string $message [, long $code [, Throwable $previous = NULL]]])

    /home/xxx/public_html/xxx/lib/private/AppFramework/App.php - line 156:

    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Gestion\ ... {}, "config")

    /home/xxx/public_html/xxx/lib/private/Route/Router.php - line 301:

    OC\AppFramework\App::main("OCA\\Gestio ... r", "config", OC\AppFramew ... {}, { _route: "g ... "})

    /home/xxx/public_html/xxx/lib/base.php - line 1000:

    OC\Route\Router->match("/apps/gestion/config")

    /home/xxx/public_html/xxx/index.php - line 36:

    OC::handleRequest()

Spowodowany Error: Wrong parameters for Doctrine\DBAL\Driver\PDO\Exception([string $message [, long $code [, Throwable $previous = NULL]]])

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Driver/AbstractException.php - line 34:

    Exception->__construct("SQLSTATE[HY ... r", null, PDOException ... ]})

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Driver/PDO/Exception.php - line 26:

    Doctrine\DBAL\Driver\AbstractException->__construct("SQLSTATE[HY ... r", "HY000", null, PDOException ... ]})

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Driver/PDO/Result.php - line 123:

    Doctrine\DBAL\Driver\PDO\Exception::new(PDOException ... ]})

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Driver/PDO/Result.php - line 65:

    Doctrine\DBAL\Driver\PDO\Result->fetchAll(2)

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Result.php - line 107:

    Doctrine\DBAL\Driver\PDO\Result->fetchAllAssociative()

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Result.php - line 326:

    Doctrine\DBAL\Result->fetchAllAssociative()

    /home/xxx/public_html/xxx/lib/private/DB/ResultAdapter.php - line 58:

    Doctrine\DBAL\Result->fetchAll(2)

    /home/xxx/public_html/xxx/lib/private/DB/PreparedStatement.php - line 67:

    OC\DB\ResultAdapter->fetchAll(2)

    /home/xxx/public_html/xxx/apps/gestion/lib/Db/Bdd.php - line 196:

    OC\DB\PreparedStatement->fetchAll(2)

    /home/xxx/public_html/xxx/apps/gestion/lib/Db/Bdd.php - line 141:

    OCA\Gestion\Db\Bdd->execSQL("INSERT INTO ... ;", [ "xxx"])

    /home/xxx/public_html/xxx/apps/gestion/lib/Controller/PageController.php - line 75:

    OCA\Gestion\Db\Bdd->checkConfig("xxx")

    /home/xxx/public_html/xxx/lib/private/AppFramework/Http/Dispatcher.php - line 217:

    OCA\Gestion\Controller\PageController->config()

    /home/xxx/public_html/xxx/lib/private/AppFramework/Http/Dispatcher.php - line 126:

    OC\AppFramework\Http\Dispatcher->executeController(OCA\Gestion\ ... {}, "config")

    /home/xxx/public_html/xxx/lib/private/AppFramework/App.php - line 156:

    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Gestion\ ... {}, "config")

    /home/xxx/public_html/xxx/lib/private/Route/Router.php - line 301:

    OC\AppFramework\App::main("OCA\\Gestio ... r", "config", OC\AppFramew ... {}, { _route: "g ... "})

    /home/xxx/public_html/xxx/lib/base.php - line 1000:

    OC\Route\Router->match("/apps/gestion/config")

    /home/xxx/public_html/xxx/index.php - line 36:

    OC::handleRequest()
```

```
Error	PHP	Error: Undefined offset: 1 at /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Driver/PDO/Exception.php#20

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Driver/PDO/Exception.php - line 20:

    OC\Log\ErrorHandler::onError(8, "Undefined offset: 1", "/home/xxx/p ... p", 20, { exception: ... "})

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Driver/PDO/Result.php - line 123:

    Doctrine\DBAL\Driver\PDO\Exception::new(PDOException ... ]})

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Driver/PDO/Result.php - line 65:

    Doctrine\DBAL\Driver\PDO\Result->fetchAll(2)

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Result.php - line 107:

    Doctrine\DBAL\Driver\PDO\Result->fetchAllAssociative()

    /home/xxx/public_html/xxx/3rdparty/doctrine/dbal/src/Result.php - line 326:

    Doctrine\DBAL\Result->fetchAllAssociative()

    /home/xxx/public_html/xxx/lib/private/DB/ResultAdapter.php - line 58:

    Doctrine\DBAL\Result->fetchAll(2)

    /home/xxx/public_html/xxx/lib/private/DB/PreparedStatement.php - line 67:

    OC\DB\ResultAdapter->fetchAll(2)

    /home/xxx/public_html/xxx/apps/gestion/lib/Db/Bdd.php - line 196:

    OC\DB\PreparedStatement->fetchAll(2)

    /home/xxx/public_html/xxx/apps/gestion/lib/Db/Bdd.php - line 141:

    OCA\Gestion\Db\Bdd->execSQL("INSERT INTO ... ;", [ "xxx"])

    /home/xxx/public_html/xxx/apps/gestion/lib/Controller/PageController.php - line 75:

    OCA\Gestion\Db\Bdd->checkConfig("xxx")

    /home/xxx/public_html/xxx/lib/private/AppFramework/Http/Dispatcher.php - line 217:

    OCA\Gestion\Controller\PageController->config()

    /home/xxx/public_html/xxx/lib/private/AppFramework/Http/Dispatcher.php - line 126:

    OC\AppFramework\Http\Dispatcher->executeController(OCA\Gestion\ ... {}, "config")

    /home/xxx/public_html/xxx/lib/private/AppFramework/App.php - line 156:

    OC\AppFramework\Http\Dispatcher->dispatch(OCA\Gestion\ ... {}, "config")

    /home/xxx/public_html/xxx/lib/private/Route/Router.php - line 301:

    OC\AppFramework\App::main("OCA\\Gestio ... r", "config", OC\AppFramew ... {}, { _route: "g ... "})

    /home/xxx/public_html/xxx/lib/base.php - line 1000:

    OC\Route\Router->match("/apps/gestion/config")

    /home/xxx/public_html/xxx/index.php - line 36:

    OC::handleRequest()
```

`Error | PHP | Directive 'allow_url_include' is deprecated at Unknown#0`

Apps:

Nextcloud 22.0.0
Gestion 1.0.4

PHP 7.4

**Comment by baimard** [comment]
Could you please tell me information about your database ? Also, the database structure of the tables of my application ?

Maybe this is oc_gestion_xxx

**Comment by Valdnet** [comment]
I am using phpMyAdmin version 4.9.4.
There are 6 tables in the database for `oc_gestion`.

**Comment by baimard** [comment]
do you have the mysql logs when the error occurs ?

**Comment by Valdnet** [comment]
> do you have the mysql logs when the error occurs ?

I do not have.

**Comment by baimard** [comment]
Try to had this in your /etc/mysql/conf.d/nameofconffile.cnf

```
[mysqld]
sort_buffer_size=256k

**Comment by baimard** [comment]
in lib/Db/Bdd.php

replace function line 137 :
by
```
    /**
     * Check
     * TODO Translation
     */
    public function checkConfig($idNextcloud){
        $sql = "SELECT count(*) as res FROM `".$this->tableprefix."configuration` WHERE `id_nextcloud` = ?";
        $res = json_decode($this->execSQL($sql, array($idNextcloud)));
        if ( count($res) > 0 && $res[0]->res < 1 ){
            $sql = "INSERT INTO `".$this->tableprefix."configuration` (`id`, `entreprise`, `nom`, `prenom`, `siret`, `siren`, `mail`, `telephone`, `adresse`, `path`, `id_nextcloud`) VALUES (NULL, 'a remplir', 'a remplir', 'a remplir', 'a remplir', 'a remplir', 'a remplir', 'a remplir', 'a remplir', '/', ?);";
            $this->execSQL($sql, array($idNextcloud));
        }
        return $res;
    }

**Comment by Valdnet** [comment]
After replacing line 137 in` lib/Db Bdd.php` after starting the application, an internal server error occurs immediately.

**Comment by baimard** [comment]
could you please share Bdd.php code ?

**Comment by Valdnet** [comment]
Here's a piece of code with the proposed changes:

```
    /**
     * DELETE
     */
    public function gestion_delete($table, $id, $idNextcloud){
        if(in_array($table, $this->whiteTable)){
            $sql = "DELETE FROM ".$this->tableprefix.$table." WHERE `id` = ? AND `id_nextcloud` = ?";
            return $this->execSQL($sql, array($id, $idNextcloud));
        }
        return false;
    }

    /**
     * Check
     * TODO Translation
     */
    public function checkConfig($idNextcloud){
        $sql = "SELECT count(*) as res FROM `".$this->tableprefix."configuration` WHERE `id_nextcloud` = ?";
        $res = json_decode($this->execSQL($sql, array($idNextcloud)));
        if ( count($res) > 0 && $res[0]->res < 1 ){
            $sql = "INSERT INTO `".$this->tableprefix."configuration` (`id`, `entreprise`, `nom`, `prenom`, `siret`, `siren`, `mail`, `telephone`, `adresse`, `path`, `id_nextcloud`) VALUES (NULL, 'a remplir', 'a remplir', 'a remplir', 'a remplir', 'a remplir', 'a remplir', 'a remplir', 'a remplir', '/', ?);";
            $this->execSQL($sql, array($idNextcloud));
        }
        return $res;
    }

    public function isConfig($idNextcloud){
        $sql = "SELECT count(*) as res FROM `".$this->tableprefix."configuration` WHERE `id_nextcloud` = ?";
        $res = json_decode($this->execSQL($sql, array($idNextcloud)))[0]->res;
        if ( $res < 1 ){
            return false;
        }else{
            return true;
        }

    }

```

**Comment by baimard** [comment]
I need to try to reproduce your environment.

Please do this steps : 

- Rollback code
- share all information about your configuration (Virtual Machine, Docker, Arch system, all version)

**Comment by Valdnet** [comment]
The data I could get was:

Linux 5.4.17-2102.203.6.el7uek.x86_64 x86_64
Apache 2.4.39
Perl 5.16.3
PHP 7.4.13 
mysql 5.7.30

**Comment by baimard** [comment]
Sorry, I have no bug, I just can recommend to update your installation.

**Comment by simple-it** [comment]
Hello,
I ve the same problem.
When I click on any link like "devis", "facture", "produits" or in "configuration" panel on "Mon Entreprise"
![Gestion_links](https://user-images.githubusercontent.com/55109662/131510829-cc5424e2-3344-4dc2-9897-7c0db23dd469.png)

my browser is redirected to a wrong url:
from https://192.168.0.212/index.php/apps/gestion/ on click on "devis" redircted to https://192.168.0.212/apps/gestion/devis

if I add manually "/index.php" after the IP address on the secont URL it work fine.

**Comment by baimard** [comment]
Fix, in last commit, I will deploy a new version of the application tomorrow, v1.1.0

**Comment by simple-it** [comment]
Hello,

Thank you for your fast action.
The version has resolved the first problem, but some other links ve the same comportment like button "OPEN" under Quote or under Invoice in this print screen
![2021-09-06 15_08_06-Window](https://user-images.githubusercontent.com/55109662/132223052-dd3ff123-31b5-4c1b-af52-97e442690152.png)

**Comment by baimard** [comment]
Ok, it was just a "test", I know where is the problem now.

I need to modifie all the javascript code with the url generator provide by nextcloud ...

I will do this quick ! Thank you for your feedback !

**Comment by simple-it** [comment]
Hello,

I don't know if is related to this url generator issue, 

When I click on the house icon
![2021-09-07 11_07_21-Gestion - Nextcloud](https://user-images.githubusercontent.com/55109662/132318247-9dae5a84-a389-45bf-9594-e869ee34e196.png)
I'm forwarded to this :
![2021-09-07 11_06_16-Index of _apps_gestion](https://user-images.githubusercontent.com/55109662/132318338-37d19d72-0b56-4269-8da8-54ba12d67dce.png)


let me know if you want a new Issue fort that

**Comment by baimard** [comment]
I confirm that is related.

Also, I think you have a security issue in your installation. I suggest that you review your .htaccess file.

**Comment by bdrayer** [comment]
Hi,

It seems that, due to the modification, I have some problems with the links inside Gestion.

**Comment by baimard** [comment]
> It seems that, due to the modification, I have some problems with the links inside Gestion because it duplicate the name of the root directory for nextcloud inside the links : https://portal.toto.ch/files/files/index.php/apps/gestion/devis (where the directory "files" is duplicated and it shouln't be).

Where do you have this link ?

**Comment by bdrayer** [comment]
It's the link when I press the Devis button but it do the same when I click on "Mon entreprise", Facture, ... And if I remove the duplicate "files" folder I arrived on the right page.

We can speak in franch if you prefer :-)

**Comment by baimard** [comment]
Is my english so bad? ah ah

It's very strange, could you please share me your setup  ?

**Comment by bdrayer** [comment]
No but as it seems that your speak french and I also speak french it will be better I think :-)

What do you need for informations? 

I am using version 22.1.1 of Nextcloud, apache 2.4.29, php 7.4 and MariaDB 10.5.11

**Comment by baimard** [comment]
I will serch the bugs. Thank you for repporting !

**Comment by bdrayer** [comment]
Thanks to you. If you need access to my installation I can create an account if needed

**Comment by baimard** [comment]
@bdrayer est ce que ton installation est derrière un reverse proxy où il y a plusieurs vhost ?

**Comment by baimard** [comment]
@bdrayer and @simple-it I made an update yesterday.

Is it possible to check directly from the repository if the app work for you ?

**Comment by simple-it** [comment]
yes I can but I don't know how to

**Comment by baimard** [comment]
I will push the modification in a new version tomorrow. Or this night.

**Comment by baimard** [comment]
I find it !

**Comment by baimard** [comment]
solved in 1.2.2
